torchtune/models/llama3_2_vision/_model_builders.py

```diff
@@ -75,47 +123,164 @@ def llama3_2_vision_11b(
-        prompt_template (Optional[_TemplateType]): optional specified prompt template.
-            If a string, it is assumed to be the dotpath of a :class:`~torchtune.data.PromptTemplateInterface`
-            class. If a dictionary, it is assumed to be a custom prompt template mapping role to the
-            prepend/append tags.
-    
+
     Returns:
-        Llama3VisionTransform: Instantiation of the Llama 3.2 vision transform
+        DeepFusionModel: Instantiation of Llama3.2 vision model with LoRA applied to
+        a subset of the attention projections in each layer.
+
     """
-    special_tokens = parse_hf_tokenizer_json(special_tokens_path) if special_tokens_path is not None else None
-    template = _get_prompt_template(prompt_template) if prompt_template is not None else None
-    return Llama3VisionTransform(
-        path=path,
-        special_tokens=special_tokens,
+    decoder_type = LoRATrainable(decoder_trainable.lower())
+    encoder_type = LoRATrainable(encoder_trainable.lower())
+    fusion_type = LoRATrainable(fusion_trainable.lower())
+    encoder = lora_llama3_2_vision_encoder(
+        encoder_lora=encoder_type == LoRATrainable.LORA,
+        fusion_lora=fusion_type == LoRATrainable.LORA,
+        lora_attn_modules=lora_attn_modules,
+        apply_lora_to_mlp=apply_lora_to_mlp,
+        apply_lora_to_output=apply_lora_to_output,
+        patch_size=14,
+        num_heads=16,
+        clip_embed_dim=1280,
+        clip_num_layers=32,
+        clip_hidden_states=[3, 7, 15, 23, 30],
+        decoder_embed_dim=4096,
+        num_layers_projection=8,
         tile_size=image_size,
+        max_num_tiles=4,
+        in_channels=3,
+        lora_rank=lora_rank,
+        lora_alpha=lora_alpha,
+        lora_dropout=lora_dropout,
+        use_dora=use_dora,
+        quantize_base=quantize_base,
+    )
+    decoder = lora_llama3_2_vision_decoder(
+        decoder_lora=decoder_type == LoRATrainable.LORA,
+        fusion_lora=fusion_type == LoRATrainable.LORA,
+        lora_attn_modules=lora_attn_modules,
+        apply_lora_to_mlp=apply_lora_to_mlp,
+        apply_lora_to_output=apply_lora_to_output,
+        vocab_size=128_256,
+        num_layers=32,
+        fusion_interval=4,
+        num_special_tokens=8,
+        num_heads=32,
+        num_kv_heads=8,
+        embed_dim=4096,
+        max_seq_len=131_072,
+        encoder_max_seq_len=128_080,  # 20*6404
+        rope_base=500000.0,
+        intermediate_dim=14336,
+        lora_rank=lora_rank,
+        lora_alpha=lora_alpha,
+        lora_dropout=lora_dropout,
+        use_dora=use_dora,
+        quantize_base=quantize_base,
+    )
+    return DeepFusionModel(
+        encoder=encoder,
+        decoder=decoder,
+        encoder_trainable=encoder_type != LoRATrainable.FROZEN,
+        decoder_trainable=decoder_type != LoRATrainable.FROZEN,
+        fusion_trainable=fusion_type != LoRATrainable.FROZEN,
+    )
+
+
+def llama3_2_vision_90b(
```

Comment:
copied from 11b. updated docstring + a couple of parameters.


11b is this:
```
decoder = llama3_2_vision_decoder(
        vocab_size=128_256,
        num_layers=32,
        fusion_interval=4,
        num_special_tokens=8,
        num_heads=32,
        num_kv_heads=8,
        embed_dim=4096,
        max_seq_len=131_072,
        encoder_max_seq_len=128_080,  # 20*6404
        rope_base=500000.0,
        intermediate_dim=14336,
    )
 ```
  90b is this:
```
decoder = llama3_2_vision_decoder(
        vocab_size=128_256,
        num_layers=100,
        fusion_interval=4,
        num_special_tokens=8,
        num_heads=64,
        num_kv_heads=8,
        embed_dim=8192,
        max_seq_len=131_072,
        encoder_max_seq_len=128_080,  # 20*6404
        rope_base=500000.0,
        intermediate_dim=28672,
    )
 ```
 
 encoder is the same, except for decoder_embed_dim, which is 8192 instead of 4096. 
 
 Values taken from here: https://huggingface.co/meta-llama/Llama-3.2-90B-Vision/blob/main/config.json